accepted/future-releases/3009-inference-using-bounds/feature-specification.md

```diff
@@ -0,0 +1,319 @@
+Note that when `X` isn't F-bounded and its bound doesn't refer to other type variables from the same declaration, the suggested changes for step 5 of CSSTV work exactly as the unmodified step 5 of the same algorithm.
+
+## Re-evaluating the motivating example
+Let's see how type inference, updated as proposed, works in the motivating example.
+
+The downwards phase isn't affected, since it doesn't provide any contextual type information in this specific example. In the upwards phase, before step 5 of CSSTV we have the partial solution `X = _`, the constraint set `{C <: X}`, and the solution of the constraint set for `X` is `C`, which is known (that is, it doesn't contain `_`).
+
+At step 5 of the modified CSSTV, `C <# X` is a lower-bound constraint of `X`, so we run [Subtype constraint generation][subtype-constraint-generation] with `P = C`, `Q = A<X>` (where `A<X>` is the bound of `X`), `L = {X}`. The next steps are as follows: `C <# A<X>` → `A<B> <# A<X>` → `B <# X`, which is the new constraint to add.
+
+We add `B <: X` to the constraint set for `X` (or, equivalently, merge it with the already merged constraint set for `X`, depending on the implementation) — it gives us `{C <# X, B <# X}`, and the solution for `X` becomes `X = B` at the end of step 5 of CSSTV.
+
+`X = B` is "frozen" since B is known, and the overall solution for `X` is `B`, making the motivating example compile and run without errors.
+
+The picture below shows the details of the updated type inference process for the example.
+
+```mermaid
+block-beta
+  columns 10
+  space downwards["Downwards phase"]:4 upwards["Upwards phase"]:4 space
+
+  space:5
+  newConstraint["B <: X"]
+  oldConstraint["C <: X"]
+  space:3
+
+  overallInput["Input"]
+  space
+  csstvDownwards["CSSTV"]
+  space:4
+  csstvUpwards["CSSTV"]
+  space
+  overallOutput["Output"]
+
+  space:10
+
+  space:5
+  constraintGeneration["C <# A&lt;X&gt; (A&lt;B&gt; <# A&lt;X&gt;)"]
+  space:4
+
+  space:2
+  boundsDownwards["X extends A<X>"]
+  space:4
+  boundsUpwards["X extends A <X>"]
+  space:2
+
+  downwards -- "∅" --> csstvDownwards
+  upwards -- "C <: X" --> csstvUpwards
+
+  overallInput -- "{X = _}" --> csstvDownwards
+  csstvDownwards -- "{X = _}" --> csstvUpwards
+  csstvUpwards -- "{X = B}" --> overallOutput
+
+  oldConstraint --> constraintGeneration
+  constraintGeneration --> newConstraint
+  boundsUpwards --> constraintGeneration
+
+  boundsDownwards --> csstvDownwards
+  boundsUpwards --> csstvUpwards
+```
+
+## More examples enabled by the current proposal
+
+In addition to the motivating examples, both the original and the more complex, being unbroken by the current proposal, some cases of non F-bounded type variables are enabled by it as well.
```

Comment:
```suggestion
In addition to the motivating examples, both the original and the more complex, being handled by this feature, some cases of type variables that aren't F-bounded are handled as well.
```
Unbroken is a nice word, especially if it exists, but "being handled" is probably safer. ;-)

Reply:
Agreed! Thanks for the suggestion. I've updated the file accordingly.